src/main/java/org/spdx/sbom/gradle/utils/SpdxDocumentBuilder.java

```diff
@@ -375,9 +364,25 @@ private Optional<SpdxPackage> createMavenModulePackage(
       spdxPkgBuilder.setChecksums(List.of(checksumSha1, checksumSha256));
       spdxPkgBuilder.setFilesAnalyzed(false);
 
-      return Optional.of(spdxPkgBuilder.build());
+      results.add(spdxPkgBuilder.build());
     }
-    return Optional.empty();
+    return results;
+  }
+
+  private static Map<ComponentIdentifier, Collection<File>> fromResolvedArtifacts(
+      Map<ComponentArtifactIdentifier, File> artifacts) {
+    Map<ComponentIdentifier, Collection<File>> results = new HashMap<>();
+    artifacts.forEach(
+        (identifier, file) -> {
+          if (results.containsKey(identifier.getComponentIdentifier())) {
```

Comment:
This section can be compressed down to:
```java
results.getOrDefault(identifier.getComponentIdentifier(), new ArrayList<File>()).add(file)
```

Reply:
But `getOrDefault` does not put the defaultValue back into `Map`

Reply:
oh right, yeah you have to add it in. oops

Reply:
I think computeIfAbsent is what I actually wanted

```
results.computeIfAbsent(identifier.getComponentIdentifier(), l -> new ArrayList<File>()).add(file)
```